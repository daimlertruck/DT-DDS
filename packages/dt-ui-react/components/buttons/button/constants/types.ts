export type ButtonVariant = 'contained' | 'outlined' | 'text';

export type OutlinedColors = 'neutral';

export type ContainedColors = 'primary' | 'error';

export type TextColors = 'primary';
