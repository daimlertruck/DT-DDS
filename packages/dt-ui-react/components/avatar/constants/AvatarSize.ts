export const AvatarSize = {
  Small: 'small',
  Medium: 'medium',
  Large: 'large',
} as const;

export type AvatarSize = (typeof AvatarSize)[keyof typeof AvatarSize];
