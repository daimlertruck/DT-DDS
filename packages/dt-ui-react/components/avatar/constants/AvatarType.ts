export enum AvatarType {
  Profile = 'profile',
  TeamMember = 'teamMember',
}
