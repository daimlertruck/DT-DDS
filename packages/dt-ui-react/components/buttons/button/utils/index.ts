export * from './buttonVariantColors';
export * from './buttonVariantStyles';
