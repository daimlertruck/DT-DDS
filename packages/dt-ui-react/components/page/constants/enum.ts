export enum PageType {
  None = 'none',
  Detail = 'detail',
}
