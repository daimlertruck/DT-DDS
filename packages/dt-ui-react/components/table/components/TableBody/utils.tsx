import { Children, ReactNode, cloneElement, isValidElement } from 'react';

import { TableDataCellHeader } from '../TableDataCell/TableDataCell';

export const addHeadersToDataCells = (
  dataCells: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(dataCells, (dataCell, index) => {
    if (isValidElement(dataCell)) {
      const header = headers[index];
      return cloneElement(
        dataCell,
        {},
        <>
          <TableDataCellHeader>{header}</TableDataCellHeader>
          {dataCell.props.children}
        </>
      );
    }
    return dataCell;
  });
};

export const addHeadersToDataCellsInRows = (
  rows: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(rows, (row) => {
    if (isValidElement(row)) {
      return cloneElement(
        row,
        {},
        addHeadersToDataCells(row.props.children, headers)
      );
    }
    return row;
  });
};
