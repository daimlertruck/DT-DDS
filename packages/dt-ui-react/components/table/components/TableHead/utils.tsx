import { Children, ReactNode, isValidElement } from 'react';

const extractColumnHeaders = (columnHeaders: ReactNode): ReactNode[] => {
  return Children.toArray(columnHeaders).reduce<ReactNode[]>(
    (headers, columnHeader) => {
      if (isValidElement(columnHeader)) {
        return [...headers, columnHeader.props.children];
      }

      return headers;
    },
    []
  );
};

export const extractTableHeaders = (theadRow: ReactNode) => {
  if (isValidElement(theadRow)) {
    return extractColumnHeaders(theadRow.props.children);
  }

  return [];
};
