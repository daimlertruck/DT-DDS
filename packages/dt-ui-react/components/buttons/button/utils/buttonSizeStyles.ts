import { CustomTheme as Theme } from '../../../../types';
import { ButtonSizes } from '../constants/types';

export const buttonSizeStyles = (size: ButtonSizes, theme: Theme) =>
  ({
    large: `
      ${theme.fontStyles.button1}
      padding: ${theme.spacing['2xs']} ${theme.spacing.s};
    `,
    medium: `
      ${theme.fontStyles.button2}
      padding: ${theme.spacing['3xs']} ${theme.spacing.xs};
    `,
    small: `
      ${theme.fontStyles.button2}
      padding: ${theme.spacing['4xs']} ${theme.spacing['2xs']};
    `,
  }[size]);
