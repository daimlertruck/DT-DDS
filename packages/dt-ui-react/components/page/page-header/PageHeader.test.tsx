import { render } from '@testing-library/react';
import React from 'react';

import { PageHeader, Button } from '../../..';
import { withProviders } from '../../../utils';
import { PageType } from '../constants';

describe('<PageHeader /> component', () => {
  const ProvidedPageHeader = withProviders(PageHeader);

  describe('When used in list page', () => {
    it('Should render a header element with display flex and correct given content', () => {
      const { container } = render(
        <ProvidedPageHeader dataTestId='list'>
          <PageHeader.Title>Page Title</PageHeader.Title>
        </ProvidedPageHeader>
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('When used in detail page', () => {
    it('Should render a header element with display flex and correct given childrens', () => {
      const { container } = render(
        <ProvidedPageHeader variant={PageType.Detail} dataTestId='detail'>
          <PageHeader.Segment>
            <PageHeader.Title variant={PageType.Detail}>
              Detail Page Title
            </PageHeader.Title>
          </PageHeader.Segment>
          <PageHeader.Description>
            Some optional Description
          </PageHeader.Description>
          <PageHeader.Segment isAlignToRight={true}>
            <Button onClick={() => null} dataTestId='click-me'>
              Click Me
            </Button>
          </PageHeader.Segment>
        </ProvidedPageHeader>
      );

      expect(container).toMatchSnapshot();
    });
  });
});
