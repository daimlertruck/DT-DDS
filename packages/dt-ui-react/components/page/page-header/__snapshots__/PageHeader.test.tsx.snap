// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageHeader /> component When used in detail page Should render a header element with display flex and correct given childrens 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  background-color: #FFFFFF;
  min-height: 228px;
  padding: 48px 32px 40px 32px;
  box-shadow: 1px 1px 3px 1px rgba(0, 0, 0, 0.1);
}

.emotion-2 {
  width: 100%;
}

.emotion-4 {
  font-weight: 400;
  font-size: 2rem;
  line-height: 1.23;
  margin: 0;
  color: #151617;
}

.emotion-6 {
  max-width: 50vw;
  margin-top: 16px;
}

.emotion-8 {
  font-weight: 400;
  font-size: 1.125rem;
  line-height: 1.34;
  margin: 0;
  color: #4F4F4F;
}

.emotion-10 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: right;
  -ms-flex-pack: right;
  -webkit-justify-content: right;
  justify-content: right;
  gap: 20px;
}

.emotion-12 {
  border: 0;
  text-transform: uppercase;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.34;
  margin: 0;
  font-weight: 700;
  border-radius: 3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
  white-space: nowrap;
  cursor: pointer;
  max-width: 20vw;
  padding: 0px 15px;
  height: 32px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  color: #FFFFFF;
  background-color: #0078D6;
}

.emotion-12:disabled {
  opacity: 0.4;
  cursor: default;
}

.emotion-12:disabled {
  pointer-events: none;
}

.emotion-12:hover {
  color: #FFFFFF;
  background-color: #01508E;
}

<div>
  <header
    class="emotion-0 emotion-1"
    data-testid="detail-page-header"
    direction="row"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <h1
        class="emotion-4 emotion-5"
        color="gray_300"
        data-testid="typography"
        style="overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"
      >
        Detail Page Title
      </h1>
    </div>
    <div
      class="emotion-6 emotion-7"
    >
      <p
        class="emotion-8 emotion-5"
        color="gray_200"
        data-testid="header-description"
      >
        Some optional Description
      </p>
    </div>
    <div
      class="emotion-10 emotion-3"
    >
      <button
        class="emotion-12 emotion-13"
        color="primary"
        data-testid="click-me-button"
      >
        Click Me
      </button>
    </div>
  </header>
</div>
`;

exports[`<PageHeader /> component When used in list page Should render a header element with display flex and correct given content 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-2 {
  font-weight: 700;
  font-size: 3rem;
  line-height: 1.23;
  margin: 0;
  color: #151617;
}

<div>
  <header
    class="emotion-0 emotion-1"
    data-testid="list-page-header"
    direction="row"
  >
    <h1
      class="emotion-2 emotion-3"
      color="gray_300"
      data-testid="typography"
      style="overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"
    >
      Page Title
    </h1>
  </header>
</div>
`;
